data/json/items/generic/casing.json

```diff
@@ -130,11 +130,23 @@
     "volume": "5ml"
   },
   {
-    "id": "40mm_casing",
+    "id": "40x46mm_casing",
     "type": "GENERIC",
     "category": "spare_parts",
-    "name": "40mm canister",
-    "description": "A large canister from a spent 40mm grenade.",
+    "name": "40x46mm casing",
+    "description": "A large canister from a spent 40x46mm grenade.",
+    "weight": "50 g",
+    "volume": "25ml",
```

Comment:
(20mm)^2⋅pi⋅46mm gives ~ 57805 mm^3, or 58ml.
Theoretically you could stuff them full of other things, and explain the lower volume this way.

Reply:
There are a load of different casings used in 40mm grenades and I can't find exact dimensions for them. Some of the cartridges are twice the length of others and I don't think 46mm is the length of the casing.

Reply:
For the "usual" fragrounds I have seen pictures of, base x a bit more than the base seems plausible, some very quick research tells me that the second number in the 40x46-9x19-5.7x28 pattern is the length of the case. Which means that 40mm might be a bit smaller than the case diameter. However, I have never even seen a 40mm grenade or wepon system in reality, so I should by no means be taken too seriously.

Reply:
> Some of the cartridges are twice the length of others

That is why I only suggested volumes for the cases. :)